### PR TITLE
FW-2052 Access files signed URL in web plugin [Security, Important]

### PR DESCRIPTION
**Pull request by Prochnost55**
### What do you want to achieve?
- To add security for image files.

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [x] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

- add attachment: true in kommunicate settings
- start new conversation and upload an image
- the image should upload
- go back and reopen conversation
- the image should load and upon clicking the image the modal containing image should open

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

INTRODUCTION TO THE ISSUE:
As of now all the images uploaded to aws server can directly be accessed using the url provided by them. 
To avoid this, aclsPrivate parameter is passed as true. 
Now in order to access image, we need to pass blobkey which will gives us the link for the particular image. The provided link is only accessible for 15 mins.

### CODE REVIEW

**Review by shubhamtibra — DISMISSED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 1]
```diff
@@ -4514,6 +4527,13 @@ var userOverride = {
                                 startTime: startTime,
                             });
                         }
+                    }else{
+                        var lazy_images = document.querySelectorAll('img.lazy_image');
+                        lazy_images.forEach(img => {
```
Comment (shubhamtibra) [inline webplugin/js/app/mck-sidebox-1.0.js:4532]:
```suggestion
                        lazy_images.forEach(function (img) {
```

**Review by shubhamtibra — DISMISSED** [review]

File: webplugin/js/app/kommunicate-ui.js [hunk 2]
```diff
@@ -1378,4 +1378,39 @@ KommunicateUI = {
...
+            rect.left >= targetRect.left &&
+            rect.bottom <= targetRect.bottom &&
+            rect.right <= targetRect.right
+        );
+    },
+    processLazyImage: function(img){
+        img.classList.remove('lazy_image');
+        var thumbnailBlobKey = img.getAttribute('data-thumbnailBlobKey');
+        KommunicateUI.getUrlFromBlobKey(thumbnailBlobKey, function(err, thumbnailUrl){
+            if (err) throw err;
```
Comment (devashishmamgain) [inline webplugin/js/app/kommunicate-ui.js:1409]: use { even if it is one line

File: webplugin/js/app/kommunicate-ui.js [hunk 3]
```diff
@@ -1378,4 +1378,39 @@ KommunicateUI = {
...
+            thumbnailUrl && (img.src = thumbnailUrl);
+            setTimeout(function(){
+                img.classList.add('lazy_image');
+            },15*60*1000);
```
Comment (devashishmamgain) [inline webplugin/js/app/kommunicate-ui.js:1413]: replace it with constant

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 4]
```diff
@@ -3174,14 +3174,28 @@ var userOverride = {
                 var title = kommunicateCommons.formatHtmlTag(
                     $this.data('name')
                 );
-                if (href === '') {
+                var blobKey = $this.find('img').attr('data-blobKey');
+                
+                if(blobKey && href == ''){
+                    KommunicateUI.getUrlFromBlobKey(blobKey, function(err, url){
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:3180]: is this code same as the one added in kommunicate-ui? can this be made common?

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 5]
```diff
@@ -4514,6 +4527,13 @@ var userOverride = {
                                 startTime: startTime,
                             });
                         }
+                    }else{
+                        var lazy_images = document.querySelectorAll('img.lazy_image');
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:4531]: use same naming convention as used at other places in javascript

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 6]
```diff
@@ -9006,9 +9037,10 @@ var userOverride = {
                                         kommunicateCommons.formatHtmlTag(
                                             msg.fileMeta.name
                                         ) +
-                                        '"><img src="' +
+                                        '"><img class="lazy_image" src="' +
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:9040]: i guess we are using - in css naming

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

**Review by hardikparashar — APPROVED** [review]

File: webplugin/js/app/kommunicate-ui.js (hunk 3 above)
Comment (devashishmamgain) [reply webplugin/js/app/kommunicate-ui.js:1413]: constant is not added, is it marked as resolved by mistake, or changes are not yet pushed to repo?

**Review by devashishmamgain — CHANGES_REQUESTED** [review]
couldn't find the changes related to comment which were resolved

**Comment by Prochnost55** [comment]
> couldn't find the changes related to comment which were resolved

changes weren't pushed as I was waiting for image placeholder cdn url to work.

as of now committed the changes.
will update the URL once I get it

**Review by devashishmamgain — DISMISSED** [review]

**Review by devashishmamgain — APPROVED** [review]

**Review by shubhamtibra — APPROVED** [review]

**Review by hardikparashar — APPROVED** [review]

**Pull request merged**